### PR TITLE
Remove Rename UI in favor of new WorkspaceEdit Preview API

### PR DESCRIPTION
**Pull request by Colengms**
Removes code for Rename UI. #4990
Switches to using WorkspaceEdit Preview API for rename confirmations.

image

Note that there does not appear to be a way for us to control the sort order of groups (when viewing by group).  VS Code appears to intentionally put our group containing confirmed references last.

Note that the icons for reference type are displayed only on groups, when viewing by group.

Also addresses https://github.com/microsoft/vscode-cpptools/issues/4131 , as updating to the new VS Code API's require switching to `vscode-test`.

### CODE REVIEW

**Comment by Colengms** [comment]
Still reworking the tests...

File: Extension/test/integrationTests/IntelliSenseFeatures/index.ts [hunk 1]
```diff
@@ -1,42 +1,38 @@
...
+  // Create the mocha test
+  const mocha = new Mocha({
+    ui: 'tdd'
+  });
+  mocha.useColors(true);
+
+  const testsRoot = path.resolve(__dirname, '.');
+
+  return new Promise((c, e) => {
+    glob('**/**.test.js', { cwd: testsRoot }, (err, files) => {
```
Comment (bobbrow) [inline Extension/test/integrationTests/IntelliSenseFeatures/index.ts]: nit: There seems to be inconsistent spacing in the tests.  I'm seeing 2-space and tab indented files.  Can you normalize them all to use 4-space indents?

Comment (Colengms) [reply Extension/test/integrationTests/IntelliSenseFeatures/index.ts]: FYI, the test runner (new index.ts) and runTest.ts were boiler-plate files provided by the vscode-test migration documentation.  I could change them, but there would also be value in keeping them as similar to the boiler-plate files as possible.

**Comment by sean-mcmanus** [comment]
Can we get VS Code to add some sort order so we can force the Confirmed Reference type to come first? Otherwise, that seems like a major UI regression, since the confirmed references are the most important ones and could otherwise be scrolled off the screen.

**Comment by Colengms** [comment]
> Can we get VS Code to add some sort order so we can force the Confirmed Reference type to come first? Otherwise, that seems like a major UI regression, since the confirmed references are the most important ones and could otherwise be scrolled off the screen.

They appear to be intentionally putting confirmed groups last, so it may be by-design.  Perhaps the most important ones are the ones that need attention (unconfirmed), so perhaps this is better?  Since this is a shipped feature, perhaps open an issue in their repo?

**Comment by sean-mcmanus** [comment]
> Can we get VS Code to add some sort order so we can force the Confirmed Reference type to come first? Otherwise, that seems like a major UI regression, since the confirmed references are the most important ones and could otherwise be scrolled off the screen.
> 
> They appear to be intentionally putting confirmed groups last, so it may be by-design.  Perhaps the most important ones are the ones that need attention (unconfirmed), so perhaps this is better?  Since this is a shipped feature, perhaps open an issue in their repo?

I filed issue https://github.com/microsoft/vscode/issues/94210

File: Extension/package.json [hunk 2]
```diff
@@ -1719,21 +1527,22 @@
     "minimatch": "^3.0.4",
     "minimist": "^1.2.5",
     "mkdirp": "^0.5.1",
+    "mocha": "4",
```
Comment (sean-mcmanus) [inline Extension/package.json]: Why isn't this something like "^4.0.0"?

File: Extension/src/LanguageServer/referencesModel.ts [hunk 3]
```diff
@@ -6,35 +6,19 @@
...
-        } else {
-            results = resultsInput.referenceInfos.filter(r => r.type !== ReferenceType.Confirmed);
-        }
+        let results: ReferenceInfo[] = resultsInput.referenceInfos.filter(r => r.type !== ReferenceType.Confirmed);
 
         // Build a single flat list of all leaf nodes
-        // Currently, the hierachy is build each time referencesTreeDataProvider requests nodes.
-        // When moving between pending and candidate views in rename, the hierachy gets rebuilt.
-        // TODO: If that is a performance issue, we could change this to always build the tree up front.
+        // Currently, the hierachy is built each time referencesTreeDataProvider requests nodes.
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/referencesModel.ts]: hierarchy

File: Extension/src/LanguageServer/referencesModel.ts [hunk 4]
```diff
@@ -78,15 +61,13 @@ export class ReferencesModel {
         return result;
     }
 
-    getFileNodes(refType: ReferenceType | undefined, isRenameCandidate: boolean): TreeNode[] {
+    getFileNodes(refType: ReferenceType | undefined): TreeNode[] {
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/referencesModel.ts]: Can this be refType? now?

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]
...

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**